levels/lovelace_tower/events.js

```diff
@@ -159,9 +159,19 @@ module.exports = async function (event, world) {
     );
   }
 
+  // Set obj4Complete and make door no longer interactable
+  if (world.isObjectiveCompleted("api-04-remote-and-local")) {
+    worldState.obj4Complete = true;
+    world.forEachEntities(`library-door-locked`, (libraryDoor) => {
+      libraryDoor.interactable = false;
+      libraryDoor.disableBody = true;
+      libraryDoor.key = "";
+    })
+  }
```

Comment:
The challenge here is that we still want the door to be interactable until the player has cast the spell on it. By disabling interaction as soon as the api-04 objective is complete, the player won't have the exclamation point appear above their head which helps indicate that it's possible